src/main/java/com/google/sps/servlets/FavouriteServlet.java

```diff
@@ -0,0 +1,118 @@
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that sends and gets user's favourite recommendation */
+@WebServlet("/favourite")
+public class FavouriteServlet extends HttpServlet {
+
+  private DatastoreService datastore;
+  private UserService userService;
+
+  @Override
+  public void init(){
+    datastore = DatastoreServiceFactory.getDatastoreService();
+    userService = UserServiceFactory.getUserService();
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get list of user's favourite recommendations from datastore
+    String email = userService.getCurrentUser().getEmail();
+    Query query = new Query("Favourite").setFilter(new FilterPredicate("email", FilterOperator.EQUAL, email));
+    PreparedQuery prepared = datastore.prepare(query);
+
+    // Get recommendations based on recommendation ID as key
+    List<Recommendation> favouriteRecommendations = new ArrayList<>();
+    for (Entity favourite : prepared.asIterable()) {
+      long recommendationId = (long) favourite.getProperty("recommendation-id");
+      Key recommendationKey = KeyFactory.createKey("Recommendation", recommendationId);
+      try {
+        Entity recommendation = datastore.get(recommendationKey);
+        favouriteRecommendations.add(convertEntityToRecommendation(recommendation));
+      }
+      catch (EntityNotFoundException e) {
+        e.printStackTrace();
+        response.sendError(HttpServletResponse.SC_NOT_FOUND);
+      }
+    }
+
+    // Return list of favourite recommendations
+    String json = new Gson().toJson(favouriteRecommendations);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String email = userService.getCurrentUser().getEmail();
+    long recommendationId = Long.parseLong(request.getParameter("id"));
+
+    // Send favourited recommendation ID to datastore, if it has not been stored
+    if(!isFavourite(email, recommendationId)) {
+      Entity favourite = new Entity("Favourite");
+      favourite.setProperty("email", email);
+      favourite.setProperty("recommendation-id", recommendationId);
+      datastore.put(favourite);
+    }
+  }
+
+  private boolean isFavourite(String email, long recommendationId) {
+    Query query = new Query("Favourite")
+      .setFilter(new FilterPredicate("email", FilterOperator.EQUAL, email))
+      .setFilter(new FilterPredicate("recommendation-id", FilterOperator.EQUAL, recommendationId));
+    List<Entity> favourite = datastore.prepare(query).asList(FetchOptions.Builder.withDefaults());
+    return favourite != null && !favourite.isEmpty();
```

Comment:
nit: you still can use `asIterable()` here so you won't need to provide the default `FetchOptions`.
Then you can check whether the iterator [`hasNext()`](https://docs.oracle.com/javase/8/docs/api/java/util/Iterator.html#hasNext--) instead if `isEmpty()`